[M5G-779] Translated languages on announcement bubbles, Pitbull edition

ashleyqw opened the pull request:
# Jira: 
https://clever.atlassian.net/browse/M5G-779
https://www.figma.com/file/X5yHaqtEAyCAtjBFeXRb2s/Messages-v2-(Translations)?node-id=625%3A18469

# Overview:
Reworking the translations tooltip in NormalAnnouncementBubble a bit to match new designs. Now passing in the actual list of languages (with their English name) to the bubble, so that we can display up to 3 of the languages in the tooltip. Also passing in a prop for an onClick function for the languages icon. Also adds details like turning blue on hover, getting the clicky hand cursor on hover, aligning the icon with the text, and fixing a bug with the read receipts icon on launchpad mobile.

# Screenshots/GIFs:


https://user-images.githubusercontent.com/35714960/135170121-748f8174-bea2-4dfa-adb4-26b80534ccd1.mov

- [x] Also tested on mobile and desktop in actual launchpad
- [x] Checked that read receipts onMouseEnter hover function still works


- [ ] Unit tests
- Manual tests:
  - [x] Chrome
  - [ ] Safari
  - [ ] IE11

# Roll Out:

- Before merging:
  - [x] Updated docs
  - [x] Bumped version in `package.json`
    - Breaking change?
      - If it is a beta component run `npm version minor`
      - If the component is not in beta run `npm version major`
    - New component or backward-compatible component feature change? Run `npm version minor`
    - Only changing documentation? All good. Skip this step.
  - After creating a new component, make sure to add it to the Components List in `ComponentsView.jsx`. To do so:
    - [ ] Add a screenshot of the component in `docs/assets/img` with the format `<COMPONENT URL LINK>.png`
- After merging:
  - [ ] Deployed updated docs (`make deploy-docs`)
  - [ ] Posted in #eng if I made a breaking change to a beta component

clcaelynn-clever commented on docs/components/AnnouncementBubbleView.jsx:
```diff
@@ -157,6 +160,9 @@ export default class AnnouncementBubbleView extends React.PureComponent {
               attachments={attachmentsArray}
               numTranslatedLanguages={10}
+              onTranslatedLanguagesClick={() =>
+                console.log("You clicked the language tooltip! Pitbull says dale again!")
```
"again" 👏🏻

clcaelynn-clever commented on docs/components/AnnouncementBubbleView.jsx:
```diff
@@ -125,6 +125,9 @@ export default class AnnouncementBubbleView extends React.PureComponent {
                 />
               }
               numTranslatedLanguages={1}
+              onTranslatedLanguagesClick={() =>
+                console.log("You clicked the language tooltip! Pitbull says dale!")
+              }
```
I'm glad we're using the `onTranslatedLanguagesClick` prop in the examples! Shouldn't it also go in the list of props tho?

clcaelynn-clever commented on src/AnnouncementBubble/NormalAnnouncementBubble.less:
```diff
@@ -155,8 +155,20 @@ button.NormalAnnouncementBubble--deleteMenuItem {
+    transition: none;
+  }
+
+  &:focus:not(:focus-visible) {
```
I think `:focus-visible` doesn't work well on Safari, or works differently? I tried using it elsewhere recently and there were issues. 😬 

Just suggest checking it out!

ashleyqw replied:
I've just tested on Safari and had no issues with these changes in particular. The MessagingThreadHistory is still getting outlined when we don't want it to, but everything is still getting outlined when it's supposed to. So it's a problem of too many outlines
But things are working and looking ok for these changes (and most others besides ThreadHistory!)

clcaelynn-clever reviewed (commented):
Just a couple questions/comments, looking great! 👍🏻  ✨

ashleyqw commented:
@clcaelynn-clever Ready for another look when you can! Tested on Safari and it seems all good on these changes (see comment above). Added an onTranslatedLanguagesHover prop for analytics. Updated docs.

clcaelynn-clever reviewed (approved):
Looking great! 🚀  🎊
